### PR TITLE
SCCS testing improvements

### PR DESCRIPTION
**Pull request by vladak**
This change increases test coverage of the SCCS repository type and performs cleanup of related code. This is quite important for some internal repositories.

### CODE REVIEW

File: opengrok-indexer/src/main/java/org/opengrok/indexer/history/SCCSRepository.java [hunk 1]
```diff
@@ -53,18 +52,23 @@
...
-    public static final String CMD_PROPERTY_KEY
-            = "org.opengrok.indexer.history.SCCS";
+    public static final String CMD_PROPERTY_KEY = "org.opengrok.indexer.history.SCCS";
     /**
      * The command to use to access the repository if none was given explicitly.
      */
     public static final String CMD_FALLBACK = "sccs";
 
-    private static final String CODEMGR_WSDATA = "Codemgr_wsdata";
+    static final String CODEMGR_WSDATA = "Codemgr_wsdata";
```
Comment (ahornace) [inline opengrok-indexer/src/main/java/org/opengrok/indexer/history/SCCSRepository.java:61]: Use `@VisibleForTesting` ?

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java [hunk 2]
```diff
@@ -18,54 +18,219 @@
...
         tdir.deleteOnExit();
         test.deleteOnExit();
         SCCSRepository instance = new SCCSRepository();
         assertEquals(shouldPass, instance.isRepositoryFor(tdir));
     }
 
     @Test
-    public void testIsRepositoryForCodemgr1() throws IOException {
+    void testIsRepositoryForCodemgr1() throws IOException {
         testIsRepositoryFor("Codemgr_wsdata", true);
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java:102]: Maybe use `CODEMGR_WSDATA`?

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java [hunk 3]
```diff
@@ -18,54 +18,219 @@
...
+     * Test of {@link SCCSRepository#getHistoryGet(OutputStream, String, String, String)}.
+     */
+    @ParameterizedTest
+    @MethodSource("getHistoryGetParams")
+    void testGetHistoryGet(final GetHistoryTestParams testParams) throws Exception {
+        try (InputStream inputStream = sccsRepository.getHistoryGet(repositoryRoot.toString(),
+                "main.c", testParams.revision)) {
+            assertNotNull(inputStream);
+            byte[] buffer = new byte[1024];
+            IOUtils.readFully(inputStream, buffer);
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java:226]: Maybe https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/io/InputStream.html#readAllBytes() would be better?

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java [hunk 4]
```diff
@@ -36,92 +36,55 @@
...
-            try {
-                if (p != null) {
-                    p.destroy();
-                }
-            } catch (Exception e) {
-
-            }
-        }
-    }
+    private File sccs_file;
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java:50]: Maybe follow Java naming? e.g. `sccsFile` and `sccsDir`?

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java [hunk 5]
```diff
@@ -36,92 +36,55 @@
...
+                out.flush();
             }
-            out.flush();
-            in.close();
-            out.close();
         } catch (IOException ex) {
-            if (sccsfile != null) {
-                sccsfile.delete();
-                sccsdir.delete();
+            if (sccs_file != null) {
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java:72]: Isn't `@AfterEach` executed even when the test throws exception? In that case this part shouldn't be necessary.

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java [hunk 6]
```diff
@@ -140,20 +103,18 @@ private String readInput(InputStream in) throws IOException {
...
      */
     @Test
-    public void getRevision() throws Exception {
-        if (!haveSccs) {
-            System.out.println("sccs not available. Skipping test");
-            return;
-        }
-        ZipInputStream zstream = new ZipInputStream(getClass().getResourceAsStream("/history/sccs-revisions.zip"));
+    void getRevision() throws Exception {
+        InputStream inputStream = getClass().getResourceAsStream("/history/sccs-revisions.zip");
```
Comment (ahornace) [inline opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java:110]: Use `try-with-resources`?

**Review by ahornace — APPROVED** [review]
Awesome job! :)

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java (hunk 5 above)
Comment (vladak) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSgetTest.java:72]: good point !

File: opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java (hunk 3 above)
Comment (vladak) [reply opengrok-indexer/src/test/java/org/opengrok/indexer/history/SCCSRepositoryTest.java:226]: Definitely, thanks !

**Comment by vladak** [comment]
Thanks for the review !

**Comment by vladak** [comment]
Ready for the upcoming 50 year anniversary of SCCS :-D

**Pull request merged**